### PR TITLE
molecules.cc: cosine sim + iterative

### PR DESCRIPTION
**Pull request by afshinfard**

### CODE REVIEW

**Comment by jowong4** [comment]
Can you change your variables to camel case?

File: src/physlr-molecules.cc [hunk 1]
```diff
@@ -333,100 +545,281 @@ biconnectedComponents(
...
+	for (uint64_t i = 0; i < adj_mat.size(); i++) {
+		for (uint64_t j = i + 1; j < adj_mat.size(); j++) {
+			if (cosSimilarity2d[i][j] < threshold) {
+				adj_mat[i][j] = 0;
+				adj_mat[j][i] = 0;
+			}
+		}
+	}
+
+	// 4- Detect Communities (find connected components - DFS)
```
Comment (jowong4) [inline src/physlr-molecules.cc]: factor the connected components out in my opinion.

Comment (afshinfard) [reply src/physlr-molecules.cc]: Good idea, will do that.

**Comment by jowong4** [comment]
Can you change instances of `unordered_map` and `unordered_set` to be robin hash versions. are you adding the OpenMP part later?

File: src/physlr-molecules.cc [hunk 2]
```diff
@@ -217,6 +252,176 @@ readTSV(graph_t& g, const std::vector<std::string>& infiles, bool verbose)
...
+    componentToVertexSet_t& binnedNeighbours,
+    uint64_t binSize = 50)
+{
+	// //   Iterate over each component and if its bigger than binSize:
+	// //   randomly split the component (set of vertices) into smaller even bins
+
+	std::vector<uint64_t> componentsSize;
+	uint64_t neighborhoodSize;
+	uint64_t componentsCount;
+	for (uint64_t i = 0; i < source.size(); i++) { // NOLINT
```
[the remark this answers is not in the document]
Comment (afshinfard) [reply src/physlr-molecules.cc]: There two datastructures  `source` and `componentsSize`, and their items has a 1-to-1 relation. If I use range it access the elements randomly and then they're not consistent.

**Comment by afshinfard** [comment]
> Can you change instances of `unordered_map` and `unordered_set` to be robin hash versions.

I'd prefer to postpone it to another PR

> are you adding the OpenMP part later?

It's added already, isn't it?

**Pull request merged**